llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1023,6 +1038,18 @@ InstructionCost VPWidenIntrinsicRecipe::computeCost(ElementCount VF,
   // clear Arguments.
   // TODO: Rework TTI interface to be independent of concrete IR values.
   SmallVector<const Value *> Arguments;
+
+  Intrinsic::ID FID = VectorIntrinsicID;
+  unsigned NumOperands = getNumOperands();
+  if (VPIntrinsic::isVPIntrinsic(VectorIntrinsicID)) {
+    std::optional<Intrinsic::ID> ID =
+        VPIntrinsic::getFunctionalIntrinsicIDForVP(VectorIntrinsicID);
+    if (ID) {
+      FID = ID.value();
```

Comment:
Why do you need this?

Reply:
1. After the VP instruction adds the EVL parameter through the VPTransform process, Arguments will be cleared and will fall Into BasicTTIImpl, which will check if the VP intrinsic is a type based instruction. All type based instruction will fall into the getTypeBasedIntrinsicInstrCost
2. VP Intrinsics should have the same cost as their non-vp counterpart
3. To ensure the accuracy of the CostAttrs parameters passed to TTI, such as the number of parameters, parameter type, ID, etc.

Reply:
@LiqinWeng I am aware that getCmpSelectInstrCost requires a predicate to estimate its cost. However, getIntrinsicInstrCost currently lacks a mechanism to provide this information.
Besides these, are there other vp intrinsics that necessitate their non-vp counter part for cost estimation?

Reply:
Mainly call instructions, eg llvm.smax/llvm.smin...

Reply:
for 1), shouldn't it be possible to retrieve the predicate from the intrinsic opcode? 

for 2), would it be sufficient to just don't clear the arguments for VPIntrinics, if needed for cost computations?

Reply:
for 1) Implemented in #108533
for 2) The root cause is that there is no way to get the EVL Value && mask Value at this stage, and VP Intrinsic cannot configure the correct ArgmentsSize in CostAttr
![image](https://github.com/user-attachments/assets/afe0ab91-f662-41e7-ac2f-57a28ecb8158)